### PR TITLE
fixes #211 SDK 2.1.0 Release Notes

### LINKED ISSUE #211: SDK 2.1.0 Release Notes
** Expected Behavior**
Add release notes for SDK 2.1.0

### PR DESCRIPTION
**Pull request by jeffngo**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build was run locally and any changes were pushed
- [ ] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

We do not have release notes for version 2.1.0 of the SDK
Issue Number: 211

## What is the new behavior?

Add SDK 2.1.0 Release Notes

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: docs/docs/Best_Practices/Sensitive_Data.md [hunk 1]
```diff
@@ -34,6 +34,9 @@ This tells the Delphix Engine to take special precautions with this password pro
 3. The Delphix Engine's UI and CLI will not display the password.
 4. Clients of the Delphix Engine's public API will not be able to access the password.
 
+!!! note
+  Removing a previously added password property from a field and running a [Data Migration](Glossary.md#data-migration) will expose the password in plaintext.
```
Comment (ravi-cm) [inline docs/docs/Best_Practices/Sensitive_Data.md:38]: Do we want to leave it as a note or add some recommendation - for e.g. "please make sure such properties handled as such in the new schema unless the intent is to change the property type". something like that?

File: docs/docs/Release_Notes/2.1.0/2.1.0.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+## New & Improved
+
+* Added the ability to migrate existing Lua toolkits to plugins.
+    * [Convert](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Converting_Migration_Scripts.md) any existing Lua upgrade scripts to Python migrations by using our new Lua upgrade [decorators](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Decorators.md).
+
+    More details about Lua Upgrade can be found [here](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Overview.md).
+
+* Added a new field `luaName` to the [Plugin Config](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Config.md) which allows plugins to define the name of the LUA toolkit that the plugin can upgrade.
+* Added a new field `minimumLuaVersion` to the [Plugin Config](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Config.md) which allows plugins to define the lowest LUA toolkit version that the plugin can upgrade.
```
Comment (ravi-cm) [inline docs/docs/Release_Notes/2.1.0/2.1.0.md:13]: Do we want to leave a note somewhere that these 2 fields go together and cannot leave one or the other?

File: docs/docs/Best_Practices/Sensitive_Data.md (hunk 1 above)
Comment (crystalplumage) [reply docs/docs/Best_Practices/Sensitive_Data.md:38]: Maybe something like "If a new release of your plugin removes the `password` keyword from a previously-existing property, then the Delphix engine will stop taking these precautions, even with already-exiting data. So, for example, the property will no longer be stored in encrypted form."

File: docs/docs/Release_Notes/2.1.0/2.1.0.md (hunk 2 above)
Comment (crystalplumage) [reply docs/docs/Release_Notes/2.1.0/2.1.0.md:13]: Could maybe even just collapse together, something like:
* Added a new fields `luaName` and `minimumLuaVersion` to the [Plugin Config](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Config.md). These allow the plugins to specify which Lua toolkit(s) that the plugin can upgrade from.

Comment (jeffngo) [reply docs/docs/Release_Notes/2.1.0/2.1.0.md:13]: I'll go with Tom's suggestion

File: docs/docs/Best_Practices/Sensitive_Data.md (hunk 1 above)
Comment (jeffngo) [reply docs/docs/Best_Practices/Sensitive_Data.md:38]: I'll add another sentence to tell the plugin author to write a migration for the property change, but I think it's already clear that their password will not be encrypted after their change.

File: docs/docs/Release_Notes/2.1.0/2.1.0.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## New & Improved
+
+* Added the ability to migrate existing Lua toolkits to plugins.
+    * [Convert](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Converting_Migration_Scripts.md) any existing Lua upgrade scripts to Python migrations by using our new Lua upgrade [decorators](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Decorators.md).
+
+    More details about Lua Upgrade can be found [here](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Overview.md).
+
+* Added a new field `luaName` to the [Plugin Config](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Config.md) which allows plugins define the name of the LUA toolkit that the plugin can upgrade.
+* Added a new field `minimumLuaVersion` to the [Plugin Config](/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Config.md) which allows plugins to define the lowest LUA toolkit version that the plugin can upgrade.
```
[the remark this answers is not in the document]
Comment (jeffngo) [reply docs/docs/Release_Notes/2.1.0/2.1.0.md]: I'll move it up to the convert bullet

**Pull request merged**